### ISSUE TITLE
Proper ending newlines (as required by POSIX) break the test suite

### ISSUE THREAD

**Issue by Hesulan** [marko-js/marko#446]
### Description
The autotests work correctly when the `expected.*` file is created and saved on Windows. However, Linux and OS X comply with POSIX, which requires a proper ending newline (`\n` immediately followed by the `EOF` symbol). The produced `actual.*` file doesn't have one, which causes the autotests to fail.

### Possible Fix
The autotest runner could simply ignore the presence or absence of an ending newline. Alternatively, the rendered output could be made POSIX-compliant (by adding a proper EOF newline), but Windows displays it as a blank line and allows it to be removed which could lead to confusion.

### Steps to Reproduce
1. Open an `expected.*` file from the autotest directory in any text editor
2. Do one of the following:
  a. (Linux or OS X) Save the file with no changes
  b. (Windows) Insert a blank newline at the end of the file and save the changes
3. Run the test suite

### Example
Saving `test/autotests/render/assign-tag/actual.html` on Linux with no changes and running `npm test` produces:
```
AssertionError: 'HELLO 1230' deepEqual 'HELLO 1230\n'
```

### What does Git have to say about all of this?
Git expects there to be an ending newline (because there really *should* be), and has to internally add `"\ No newline at end of file"` as a workaround, shown by github as a red symbol on the last line - go look at the commit diff for almost any file written on Windows, you'll see what I mean.

### Workarounds
The only workaround I can find is to delete the last line and use `echo -n ... >>` to reinsert it without a trailing newline.

**Comment by patrick-steele-idem** [comment]
Thank you for bringing this up. I think it makes sense to normalize the HTML string before doing the comparison to avoid issues related to having/not-having a new line character at the end:

```javascript
str = str.replace(/\s/g, '')
```

Or, possibly a little more conservative:

```javascript
str = str.replace(/\r\n|\n/g, '')
```

Thoughts?

**Comment by Hesulan** [comment]
I agree, and I feel like `/(\r\n|\n)$/` (or `/\r?\n$/`) would probably be safer. The only thing to keep in mind would be if something depends on there being a newline, but that could simply add two newlines (`\n\n`) and only the second one would be removed.

**Comment by patrick-steele-idem** [comment]
Agreed. Would you be interested in making this change to normalize the string before comparison?
